File: String/Anagram/is_anagram.py
def is_anagram(a: str, b: str):
    count = [0 for _ in range(26)]
    count2 = [0 for _ in range(26)]
    a_length, b_length = len(a), len(b)

    if a_length != b_length:
        return False

    for i in range(a_length):
        if 'a' <= a[i] <= 'z':
            count[ord(a[i]) - ord('a')] += 1
        elif 'A' <= a[i] <= 'Z':
            count2[ord(a[i]) - ord('A')] += 1
        if 'a' <= b[i] <= 'z':
            count[ord(b[i]) - ord('a')] -= 1
        elif 'A' <= b[i] <= 'Z':
            count2[ord(b[i]) - ord('A')] -= 1

    if any(count) or any(count2):
        return False

    return True


if __name__ == '__main__':
    a, b = 'Abc', 'cab'
    if is_anagram(a, b):
        print('Anagrams')
    else:
        print('Not Anagrams')

File: String/Anagram/is_anagram.cpp
#include <bits/stdc++.h>
using namespace std;

bool isAnagram(string &a, string &b){
	int count[26]={};
	if((int)a.size() != (int)b.size())return 0;
	for(int i=0;i<(int)a.size();i++){
		if('a'<=a[i]&&a[i]<='z')count[a[i]-'a']++;
		if('A'<=a[i]&&a[i]<='Z')count[a[i]-'A']++;
		if('a'<=b[i]&&b[i]<='z')count[b[i]-'a']--;
		if('A'<=b[i]&&b[i]<='Z')count[b[i]-'A']--;
	}
	for(int i=0;i<26;i++)
		if(count[i]!=0)return 0;
	return 1;

}

int main(){
	string a,b;
	a = "Abc";
	b = "cab";
	if(isAnagram(a,b)){
		cout<<"Anagrams"<<endl;
	}else cout<<"Not Anagrams"<<endl;
}
